### PR TITLE
Fix slow GetProperty component

### LINKED ISSUE #182: BHoM_UI: Explode() is faster that GetProperty
As in the header, currently it is faster to retrieve a property of an object with the Explode component than with GetProperty - not sure if that is the desired behaviour (intuitively one would say GetProperty should be faster).

### PR DESCRIPTION
**Pull request by adecler**
### Issues addressed by this PR

Closes #182

### Test files

This is the computational time before fix:

image

And after the fix:

image

### CODE REVIEW

File: BHoM_UI/Components/Engine/GetProperty.cs [hunk 1]
```diff
@@ -87,8 +87,13 @@ public bool SetOutputTypes(object result)
             if (OutputParams.Count < 1)
                 return true;
 
-            OutputParams[0].DataType = result.GetType();
-            CompileOutputSetters();
+            Type type = result.GetType();
+            if (OutputParams[0].DataType != type)
+            {
+                OutputParams[0].DataType = result.GetType();
```
Comment (pawelbaran) [inline BHoM_UI/Components/Engine/GetProperty.cs]: Could use `type` instead of `result.GetType()`

Comment (adecler) [reply BHoM_UI/Components/Engine/GetProperty.cs]: Ha, that was the initial plan. Seems like I got distracted at the wrong time there 😋

**Pull request merged**